### PR TITLE
Add benchmark for AnnulusBounds

### PR DESCRIPTION
**Pull request by paulgessinger**
I added a benchmark for the various bounds checks in `AnnulusBounds`. I'm following @HadrienG2's example in `BoundaryCheckBenchmark.cpp`, and pick out a few special points to test best-case and worst-case performance. Also included are random test points.

This indicates the covariance check is about 10x slower than the tolerance and absolute checks. (Which is not really unexpected, I just wanted to quantify)

### CODE REVIEW

**Comment by paulgessinger** [comment]
Ok, I think this time I didn't mess it up.

**Comment by asalzburger** [comment]
I like this initiative!

**Comment by asalzburger** [comment]
Why is this labeled as bug?

**Comment by paulgessinger** [comment]
> Why is this labeled as bug?

Because after having opened this PR wrongly two times before this one, I labelled it incorrectly. 😅

File: Tests/Benchmarks/AnnulusBoundsBenchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    bool isInsideTol01 = aBounds.inside(locPC, bcTol01);
+
+    bool isInsideCov = aBounds.inside(locPC, bcCov);
+
+    os << loc.x() << "," << loc.y() << "," << isInsideAbs << "," << isInsideTol0
+       << "," << isInsideTol1 << "," << isInsideTol01 << "," << isInsideCov
+       << std::endl;
+  }
+
+  // // This point is inside the area
```
Comment (asalzburger) [inline Tests/Benchmarks/AnnulusBoundsBenchmark.cpp:80]: Is this replaced by the testPoints ?

Then I would just remove the block.

Comment (paulgessinger) [reply Tests/Benchmarks/AnnulusBoundsBenchmark.cpp]: Ah yeah. Removed it.

**Pull request merged**